experimental/form-engine/README.md

```diff
@@ -0,0 +1,470 @@
+});
+
+// TOUCHED strategy (legacy) - based on touched fields
+engine.init({ email: '' }, {
+  dirtyCheckStrategy: DIRTY_CHECK_STRATEGY.TOUCHED,
+});
+```
+
+**VALUES Strategy (Recommended)**
+- `dirty: true` when any value differs from initial value
+- `dirty: false` when all values equal initial values
+- Correctly handles "undo" scenarios (value changed back to initial)
+
+**TOUCHED Strategy (Legacy)**
+- `dirty: true` when any field was touched
+- `dirty: false` when no fields were touched
+- Does NOT detect "undo" scenarios
+
+### Custom Equality Function
+
+For complex value comparisons (objects, arrays), provide a custom equality function:
+
+```jsx
+import { isEqual } from 'lodash';
+
+engine.init({ items: [] }, {
+  dirtyCheckStrategy: DIRTY_CHECK_STRATEGY.VALUES,
+  isEqual: (a, b) => isEqual(a, b), // deep equality
+});
+```
+
+## API Reference
+
+### Form Component
+
+```jsx
+<Form
+  onSubmit={(values) => void}
+  initialValues={object}
+  defaultValidateOn="blur" | "change"
+  config={{
+    dirtyCheckStrategy: DIRTY_CHECK_STRATEGY.VALUES,
+    isEqual: (a, b) => a === b,
+    // ... other FormEngine options
+  }}
+  engine={FormEngine}
+>
+  {children}
+</Form>
+```
+
+### Field Component
+
+```jsx
+<Field
+  name="fieldName"
+  validate={(value, allValues) => string | undefined}
+  validateOn="blur" | "change"
+  debounceDelay={number}
+  subscription={object}
+>
+  {({ input, meta }) => JSX}
+</Field>
+```
+
+### Hooks
+
+- `useField(name, options)` - Field state and handlers
+- `useFormState(subscription)` - Form state
+- `useWatch(name, options)` - Watch specific field with optional selector and bubble support
+- `useFormSubmit(onSubmit)` - Form submission
+
+#### useWatch Options
+
+```jsx
+// Watch specific field
+const email = useWatch('email');
+
+// Watch with selector (transform value)
+const emailLength = useWatch('email', { 
+  selector: (value) => value?.length || 0 
+});
+
+// Watch with bubble (receive nested field changes)
+const orders = useWatch('orders', { bubble: true });
+// Now updates when orders[0].amount, orders[1].status, etc. change
+
+// Combine selector and bubble
+const orderCount = useWatch('orders', { 
+  selector: (arr) => arr?.length || 0,
+  bubble: true 
+});
+```
+
+**Bubble behavior:**
+- `bubble: false` (default) - Only updates when exact field changes
+- `bubble: true` - Updates when any nested field changes
+  - Example: `useWatch('foo', { bubble: true })` receives events from `foo[0].field`, `foo.bar`, etc.
+  - Useful for watching arrays/objects and updating when ANY nested value changes
+
+### FormEngine
+
+```jsx
+const engine = new FormEngine(services);
+
+// Core methods
+engine.init(initialValues, config);
+engine.get(path);
+engine.set(path, value);
+engine.validateAll();
+engine.submit(onSubmit);
+
+// Event subscription with bubble support
+engine.on(event, callback, context, options);
+// options.bubble - if true, listener receives parent path events
```

Comment:
s/parent path events/events from nested fields/

"Parent path" gives the wrong impression here. IIUC, passing `options.bubble: true` on a field like `foo` means the listener for `foo` will also receive events from `foo.bar` or `foo[1].bat` etc. If you don't like "nested", maybe "children" or "subfields"?